bindings/go/src/fdb/database.go

```diff
@@ -61,15 +61,21 @@ type DatabaseOptions struct {
 }
 
 // Close will close the Database and clean up all resources.
-// You have to ensure that you're not resuing this database.
+// It must be called exactly once for each created database.
+// You have to ensure that you're not reusing this database
+// after it has been closed.
```

Comment:
Should we return an error in the close method if the DB is already closed?

Reply:
In that case the `Close()` function should set `d.ptr` to `nil`, which it currently doesn't, and calling `Close()` on an improperly-constructed `Database` would have an undetermined effect (most likely SIGSEGV) instead of current behavior (no-op).

I can make the change and add some test coverage, however that likely means that in future when adding `Close()` for futures and transactions they'd also need to have this return value for consistency, and that might clutter a bit the calling code.

Since having this error on `Close()` still does not do anything regarding concurrency-safety (caller must make sure of that), we could leave it without `error`; let me know what you prefer.

The error return value can always be added later in its own PR.